File: tools/df-save-re/ghidra_scripts/find_read_file.py
# Ghidra postScript: locate read_file vtable entries for target RTTI types.
# @category DF
# @keybinding
# @menupath
# @toolbar

from ghidra.program.model.symbol import SourceType

TARGETS = [
    "history_figurest",
    "hf_profilest",
    "artistic_profilest",
    "histfig_entity_linkst",
    "history_eventst",
    "world_sitest",
    "historical_entityst",
]

strings = currentProgram.getMemory()
for name in TARGETS:
    found = findBytes(None, name.encode("ascii"), 1)
    if found:
        printf("RTTI string %s @ %s\n" % (name, found))

File: tools/df-save-re/ghidra_scripts/FindReadFile.java
// Ghidra postScript: locate RTTI strings for DF save read_file targets.
//@category DF

import ghidra.app.script.GhidraScript;
import ghidra.program.model.address.Address;

public class FindReadFile extends GhidraScript {
    private static final String[] TARGETS = {
        "history_figurest",
        "hf_profilest",
        "artistic_profilest",
        "art_image_elementst",
        "art_image_propertyst",
        "histfig_entity_linkst",
        "history_eventst",
        "world_sitest",
        "historical_entityst",
    };

    @Override
    public void run() throws Exception {
        for (String name : TARGETS) {
            Address found = find(name);
            if (found != null) {
                println("RTTI string " + name + " @ " + found);
            } else {
                println("RTTI string " + name + " NOT FOUND");
            }
        }
    }
}
